src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala

```diff
@@ -903,26 +960,37 @@ trait AppAndGroupFormats {
     * In the past, healthCheck.portIndex was required and had a default value 0. When we introduced healthCheck.port, we
     * made it optional (also with ip-per-container in mind) and we have to re-add it in cases where it makes sense.
     */
+  private[this] def addHealthCheckPortIndexIfNecessary(healthChecks: Set[_ <: HealthCheck]): Set[_ <: HealthCheck] =
+    healthChecks.map {
+      case healthCheck: MarathonTcpHealthCheck =>
+        def needsDefaultPortIndex = healthCheck.port.isEmpty && healthCheck.portIndex.isEmpty
+        if (needsDefaultPortIndex) healthCheck.copy(portIndex = Some(0))
+        else healthCheck
+      case healthCheck: MarathonHttpHealthCheck =>
+        def needsDefaultPortIndex = healthCheck.port.isEmpty && healthCheck.portIndex.isEmpty
+        if (needsDefaultPortIndex) healthCheck.copy(portIndex = Some(0))
+        else healthCheck
+      case healthCheck: MesosTcpHealthCheck =>
+        def needsDefaultPortIndex = healthCheck.port.isEmpty && healthCheck.portIndex.isEmpty
+        if (needsDefaultPortIndex) healthCheck.copy(portIndex = Some(0))
+        else healthCheck
+      case healthCheck: MesosHttpHealthCheck =>
+        def needsDefaultPortIndex = healthCheck.port.isEmpty && healthCheck.portIndex.isEmpty
```

Comment:
Looks like 4 times the same functionality besides the copy function. Perhaps extract this to a separate method:
```scala
def withPort[T<:HealthCheck](healthCheck:T, adapt: T=>T): T = {
  def needsDefaultPortIndex = healthCheck.port.isEmpty && healthCheck.portIndex.isEmpty
  if (needsDefaultPortIndex) adapt(healthCheck)  else healthCheck
}
withPort(hc, hc.copy(portIndex = Some(0))
```